javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java

```diff
@@ -278,7 +278,7 @@ public static boolean areSameDeclaredTypes(Type.ClassType t1, Type.ClassType t2)
         if (t1.tsym.name != t2.tsym.name) {
             return false;
         }
-        return t1.toString().equals(t2.toString());
+        return t1.toString().equals(t1.toString());
```

Comment:
Nice find! I wonder if this check ever makes sense - if the two `tsym.name` fields are `==`, why would the string comparison not be?
Can the whole method just do `return t1.tsym.name == t2.tsym.name`?

Reply:
They said they wanted to have a cheaper test first... I guess the second comparison is necessary, I will try the fix in another PR.

Reply:
Oh, the fix is already done in here eisop#264 and my later merge accepted the wrong version.